### PR TITLE
Split numeric field type to Integer and Decimal

### PR DESCRIPTION
**Pull request by labkey-ians**
- Split numeric field type to Integer and Decimal
- Update test to reflect change

### CODE REVIEW

**Comment by labkey-adam** [comment]
We should wait a few days on this, to make sure BTC doesn't waffle. There was some discussion on this morning's call about switching to the survey metadata approach to distinguish integer vs. decimal.

**Comment by labkey-adam** [comment]
BTC has confirmed that "integer" || "decimal" is the final decision, https://www.labkey.org/FDA%20Mobile%20App/announcements-thread.view?rowId=20081

**Comment by labkey-ians** [comment]
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=39137

Committed the changes before I realized this had already been approved. Rather than try to re-open lets do these together.

File: src/org/labkey/mobileappstudy/surveydesign/ServiceSurveyDesignProvider.java [hunk 1]
```diff
@@ -105,7 +106,14 @@ private static String getServiceUrl(Container container)
     {
         Module module = ModuleLoader.getInstance().getModule(MobileAppStudyModule.NAME);
         String value = module.getModuleProperties().get(MobileAppStudyModule.METADATA_SERVICE_BASE_URL).getEffectiveValue(container);
-        return value == null ? null : value.trim();
+        if (StringUtils.isBlank(value))
+            return null;
+        value = value.trim();
```
Comment (labkey-adam) [inline src/org/labkey/mobileappstudy/surveydesign/ServiceSurveyDesignProvider.java]: Could wrap initial assignment with StringUtils.trimToNull() and then simple null check.

File: src/org/labkey/mobileappstudy/surveydesign/ServiceSurveyDesignProvider.java [hunk 2]
```diff
@@ -105,7 +106,14 @@ private static String getServiceUrl(Container container)
...
+
+        //Allow backwards compatibility to baseUrl parameter, truncate /activity from the configured url if present Issue #39137
+        return StringUtils.endsWithIgnoreCase(value, "/" + ACTIVITY_ACTION) ?
+                value.substring(0, value.length() - ACTIVITY_ACTION.length() - 1): //Truncate the action and the preceding '/'
+                value;
```
Comment (labkey-adam) [inline src/org/labkey/mobileappstudy/surveydesign/ServiceSurveyDesignProvider.java]: StringUtils.removeEndIgnoreCase() would be handy here

**Pull request merged**